File: java/ingredients/Burger.java
package ingredients;

import java.util.ArrayList;

public class Burger {
    private ArrayList<Integer> stack;
    private Burger passenger;

    public boolean addIngredient(int ingredient) {
        if (stack.isEmpty() || getWeight() <= getStrength()) {
            stack.add(ingredient);
            return true;
        }

        return false;
    }

    public int getStrength() {
        return BaseAnimal.getStrength(stack.get(0));
    }

    public int getWeight() {
        return stack.stream().mapToInt(a -> BaseAnimal.getWeight(a)).sum();
    }

    public int getHeight() {
        // int h = stack.stream().mapToInt(a -> BaseAnimal.getHeight(a)).sum();
        // if (passenger != null) h += passenger.stack.stream().mapToInt(a -> BaseAnimal.getHeight(a)).sum();

        // return h;

        // ^deleted because ima make so kangaroo doesn't add to height
        return stack.stream().mapToInt(a -> BaseAnimal.getHeight(a)).sum();
    }

    public String topAbility() {
        return BaseAnimal.topAbility(stack.get(stack.size()-1));
    }

    public String botAbility() {
        return BaseAnimal.botAbility(stack.get(0));
    }

    public void kill() {
        stack.remove(0);
    }

    public boolean dead() {
        return stack.isEmpty();
    }

}

File: game/ingredients/burger.py
from ingredients.base_animal import BaseAnimal

class Burger:
    def __init__(self):
        self.stack = []
        self.passenger = None
    
    def add_ingredient(self, ingredient):
        if not self.stack or self.get_weight() < self.get_strength():
            self.stack.append(ingredient)
            return True

        return False
    
    def get_strength(self):
        return BaseAnimal.strength[self.stack[0]]
    
    def get_weight(self):
        return sum([BaseAnimal.weight[i] for i in self.stack])
    
    def get_height(self):
        return sum([BaseAnimal.height[i] for i in self.stack])
    
    def top_ability(self):
        return BaseAnimal.top_ability[self.stack[-1]]
    
    def bot_ability(self):
        return BaseAnimal.bot_ability[self.stack[0]]

    def kill(self):
        self.stack.pop(0)

    def dead(self):
        if not self.stack:
            return True
        
        return False